components/logins/android/src/main/java/mozilla/appservices/logins/DatabaseLoginsStorage.kt

```diff
@@ -231,14 +232,17 @@ class DatabaseLoginsStorage(private val dbPath: String) : AutoCloseable, LoginsS
     }
 
     @Throws(LoginsStorageException::class)
-    override fun importLogins(logins: Array<ServerPassword>): Long {
-        val s = JSONArray().apply {
-            logins.forEach {
-                put(it.toJSON())
-            }
-        }.toString()
-        return rustCallWithLock { raw, error ->
-            PasswordSyncAdapter.INSTANCE.sync15_passwords_import(raw, s, error)
+    override fun importLogins(logins: Array<ServerPassword>): JSONObject {
+        return readQueryCounters.measure {
```

Comment:
`readQueryCounters` doesn't seem right to me, should this be `writeQueryCounters` if anything?